### ISSUE TITLE
MissingPluginException: No implementation found for method

### ISSUE THREAD

**Issue by trungtruongpham** [flutter/flutter#46960]
Hi everyone, I'm just update my FLutter app to Flutter 1.12.13 and my app can't find method login of google_sign_in: ^4.0.14 flutter_facebook_login: ^3.0.0.  I get this issue :

```bash
D/FlutterActivity(23677): Using the launch theme as normal theme.
D/FlutterActivityAndFragmentDelegate(23677): Setting up FlutterEngine.
D/FlutterActivityAndFragmentDelegate(23677): No preferred FlutterEngine was provided. Creating a new FlutterEngine for this FlutterFragment.
W/FlutterEngine(23677): Tried to automatically register plugins with FlutterEngine (io.flutter.embedding.engine.FlutterEngine@c08eae1) but could not find and invoke the GeneratedPluginRegistrant.
D/FlutterActivityAndFragmentDelegate(23677): Attaching FlutterEngine to the Activity that owns this Fragment.
D/FlutterView(23677): Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@c08eae1
D/FlutterActivityAndFragmentDelegate(23677): Executing Dart entrypoint: main, and sending initial route: /
W/DynamiteModule(23677): Local module descriptor class for providerinstaller not found.
```

I tried flutter clean and flutter run / reinstall my app on device and even emulator but it didnt work.

**Comment by iapicca** [comment]
Hi @trungtruongpham 
could you please upgrade `google_sign_in 4.1.0`
than run 
`flutter clean`
`flutter pub cache repair`
let me know if this solve your issue
thank you

**Comment by trungtruongpham** [comment]
thanks to @iapicca  I have tried flutter clean and flutter pub cache repair and when flutter pub cache repair finished i got issue :  pub finished with exit code 69 then i tried to run my app again and it still got MissingPluginException.
Btw: i got MissingPluginException both by google_sign_in and flutter_facebook_login but both of them stuck in login method.

**Comment by xster** [comment]
Can you run `flutter packages get` and look for a GeneratedPluginRegistrant.java generated file in your project?

Can you past the contents of the file for us?

**Comment by trungtruongpham** [comment]
> Can you run `flutter packages get` and look for a GeneratedPluginRegistrant.java generated file in your project?
> 
> Can you past the contents of the file for us?
Yes, here is my GeneratedPluginRegistrant.java file 

package io.flutter.plugins;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;
import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry;

/**
 * Generated file. Do not edit.
 * This file is generated by the Flutter tool based on the
 * plugins that support the Android platform.
 */
@Keep
public final class GeneratedPluginRegistrant {
  public static void registerWith(@NonNull FlutterEngine flutterEngine) {
    ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
      io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin.registerWith(shimPluginRegistry.registrarFor("io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin"));
    flutterEngine.getPlugins().add(new io.flutter.plugins.firebaseanalytics.FirebaseAnalyticsPlugin());
      io.flutter.plugins.firebaseauth.FirebaseAuthPlugin.registerWith(shimPluginRegistry.registrarFor("io.flutter.plugins.firebaseauth.FirebaseAuthPlugin"));
    flutterEngine.getPlugins().add(new io.flutter.plugins.firebase.core.FirebaseCorePlugin());
    flutterEngine.getPlugins().add(new io.flutter.plugins.firebase.database.FirebaseDatabasePlugin());
    flutterEngine.getPlugins().add(new io.flutter.plugins.firebase.storage.FirebaseStoragePlugin());
      com.roughike.facebooklogin.facebooklogin.FacebookLoginPlugin.registerWith(shimPluginRegistry.registrarFor("com.roughike.facebooklogin.facebooklogin.FacebookLoginPlugin"));
      io.flutter.plugins.googlesignin.GoogleSignInPlugin.registerWith(shimPluginRegistry.registrarFor("io.flutter.plugins.googlesignin.GoogleSignInPlugin"));
  }
}

**Comment by xster** [comment]
umm looks like the file is right. Could you either send us a repro project or look inside your APK's dex to see if the io.flutter.plugins.GeneratedPluginRegistrant class is in the dex?

**Comment by matthew-carroll** [comment]
@trungtruongpham is this issue still happening for you? If so, can you provide a minimal project that reproduces the issue?

**Comment by Lucas15andrade** [comment]
I had the same problem with cloud_firestore. When opening the subproject in Android Studio he suggested updating the version of the gradle, only after that the project started to compile normally. The strange thing that the error started to happen without any change in the project.

**Comment by iapicca** [comment]
Could everyone who still has this problem please file a new issue with the exact descriptions what happens, logs and the output of 'flutter doctor -v'  please.
All system setups can be slightly different so it's always better to open new issues and reference related issues.

**Comment by qxmmxi** [comment]
I had the same problem .
Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.4 19E266, locale zh-Hans-CN)
    • Flutter version 1.12.13+hotfix.9 at /Applications/flutter
    • Framework revision f139b11009 (2 weeks ago), 2020-03-30 13:57:30 -0700
    • Engine revision af51afceb8
    • Dart version 2.7.2


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/xxx/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/xxx/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.9.1

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.0.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (1 available)
    • M2 E • 741AEDR22LERB • android-arm64 • Android 6.0.1 (API 23)

**Comment by sebinemeth** [comment]
Same problem here. Updated Gradle, nothing changed. Error occures with `google_sign_in: ^4.5.1` and `flutter_facebook_login: ^3.0.0`.

```[ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: MissingPluginException(No implementation found for method logIn on channel com.roughike/flutter_facebook_login)
[ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: MissingPluginException(No implementation found for method init on channel plugins.flutter.io/google_sign_in)```

```flutter doctor -v
[√] Flutter (Channel stable, v1.17.1, on Microsoft Windows [Version 10.0.18363.836], locale hu-HU)
    • Flutter version 1.17.1 at C:\flutter
    • Framework revision f7a6a7906b (2 weeks ago), 2020-05-12 18:39:00 -0700
    • Engine revision 6bc433c6b6
    • Dart version 2.8.2

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Android
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = C:\Android
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] VS Code (version 1.45.1)
    • VS Code at C:\Users\Németh Sebi\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.10.2

[√] Connected device (1 available)
    • Mi A3 • e9d0210dba56 • android-arm64 • Android 10 (API 29)

• No issues found!```

Any more ideas? I think I've tried every solution I found, still no progress...
